### ISSUE TITLE
Add Setting: Automatic space insertion after word suggestion selection

### ISSUE THREAD

**Issue by ckennerknecht** [dslul/openboard#14]
When you select a word of the suggestion list there will always be inserted a space character after it when you continue to write, what can be annoying if you want to edit or add something to the suggested word. There should be a setting for en- /disabling this behavior.

**Comment by TheCrowned** [comment]
What about having as default behaviour the space being inserted immediately? Right now it is only inserted when the user types again, which means that if you hit backspace, you won't delete the new space, but rather the last char of the previous word. If the space was effectively inserted after every suggestion, this would work better.
An example: I write `svegli` with the goal of getting to `svegliato`. With `svegli` I get the suggestion of `sveglia`, which I hit. If I then type `to`, I end up with `sveglia to`, while if I hit backspace and then type, I get `sveglito`. 
Should be easy to fix this, no? With some directions on where to look for, I'd willingly do it :) @dslul

**Comment by TheCrowned** [comment]
(Actually a duplicate of #97)

**Comment by dslul** [comment]
I don't know, gboard works in the same way, and people used to it could find a change like this annoying.

**Comment by danielduner** [comment]
Adding a space automatically works well enough in English, but is terrible for languages with compound words. The default setting should probably remain as it is, but an option to opt-out would be very nice.

I made a local build for my own use, where I simply disabled the phantom space by changing `SpaceState.PHANTOM ` to `SpaceState.NONE` in `onPickSuggestionManually` in InputLogic.java.

If you want to take a look at making it a proper setting @TheCrowned, that would be excellent. Or if there is an interest in a PR, I might be able to figure out how to turn it into a proper setting.

https://github.com/dslul/openboard/blob/ac036f277e4d47ec6da317f2076480af22050e4b/app/src/main/java/org/dslul/openboard/inputmethod/latin/inputlogic/InputLogic.java#L328

**Comment by mranvick** [comment]
I agree, it would be really nice to have this feature as a setting. In addition, as mentioned in #97 the space does not appear before you start writing again, which is terrible UX (both surprising and frustrating).
